src/OpenSpaceToolkit/Astrodynamics/Flight/System/Dynamics/PositionDerivative.cpp

```diff
@@ -42,16 +47,24 @@ std::ostream& operator<<(std::ostream& anOutputStream, const PositionDerivative&
     return anOutputStream;
 }
 
-void PositionDerivative::applyContribution(
-    const NumericalSolver::StateVector& x, NumericalSolver::StateVector& dxdt, const Instant& anInstant
+Array<Shared<const CoordinatesSubset>> PositionDerivative::getReadCoordinatesSubsets() const
+{
+    return {CartesianVelocity::Default()};
+}
+
+Array<Shared<const CoordinatesSubset>> PositionDerivative::getWriteCoordinatesSubsets() const
+{
+    return {CartesianPosition::Default()};
+}
+
+VectorXd PositionDerivative::computeContribution(
+    [[maybe_unused]] const Instant& anInstant, const VectorXd& x, [[maybe_unused]] const Shared<const Frame>& aFrame
```

Comment:
```suggestion
    [[maybe_unused]] const Instant& anInstant, const VectorXd& x, [[maybe_unused]] const Shared<const Frame> aFrameSPtr
```

Didn't know you could do the `maybe_unused` thing. That's neat